File: frontend/apps/ios/Pantopus/Features/Mailbox/Unboxing/Components/FiledSummary.swift
//
//  FiledSummary.swift
//  Pantopus
//
//  A17.14 filed-state chrome. `FiledSummary` stacks the success
//  `FiledBanner` ("Filed to Home › Warranties" + Undo) over the collapsed
//  `FiledShots` photo summary ("4 photos saved · Originals kept in your
//  Vault"). `ScanNextCard` is the dashed launcher at the foot of the filed
//  frame ("Scan the next item"); the screen renders the AI elf + locked
//  `OcrFactsList` between them, matching the design order.
//

import SwiftUI

// MARK: - FiledSummary (banner + photo summary)

@MainActor
struct FiledSummary: View {
    let filedTo: String
    let filedSubtitle: String
    let shots: [UnboxingShot]
    let photosSavedLabel: String
    let onUndo: () -> Void
    let onViewPhotos: () -> Void

    var body: some View {
        VStack(spacing: Spacing.s3) {
            banner
            photoSummary
        }
    }

    private var banner: some View {
        HStack(spacing: Spacing.s3) {
            RoundedRectangle(cornerRadius: Radii.md, style: .continuous)
                .fill(Theme.Color.success)
                .frame(width: 38, height: 38)
                .overlay(Icon(.check, size: 20, strokeWidth: 2.4, color: .white))
                .shadow(color: Theme.Color.success.opacity(0.3), radius: 6, x: 0, y: 2)
            VStack(alignment: .leading, spacing: 1) {
                Text("Filed to \(filedTo)")
                    .font(.system(size: 13.5, weight: .bold))
                    .foregroundStyle(Theme.Color.success)
                Text(filedSubtitle)
                    .font(.system(size: 11.5, weight: .regular))
                    .foregroundStyle(Theme.Color.success)
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            Button(action: onUndo) {
                Text("Undo")
                    .font(.system(size: 10, weight: .bold))
                    .foregroundStyle(Theme.Color.success)
                    .padding(.horizontal, Spacing.s2)
                    .padding(.vertical, 3)
                    .background(Theme.Color.appSurface)
                    .clipShape(Capsule())
                    .overlay(Capsule().stroke(Theme.Color.successLight, lineWidth: 1))
            }
            .buttonStyle(.plain)
            .accessibilityIdentifier("unboxing_undo")
        }
        .padding(.horizontal, Spacing.s3)
        .padding(.vertical, Spacing.s3)
        .background(Theme.Color.successBg)
        .clipShape(RoundedRectangle(cornerRadius: Radii.xl, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: Radii.xl, style: .continuous)
                .stroke(Theme.Color.successLight, lineWidth: 1)
        )
        .accessibilityIdentifier("unboxing_filedBanner")
    }

    private var photoSummary: some View {
        Button(action: onViewPhotos) {
            HStack(spacing: Spacing.s3) {
                ZStack(alignment: .leading) {
                    ForEach(Array(shots.prefix(3).enumerated()), id: \.element.id) { index, _ in
                        stackedThumb
                            .offset(x: CGFloat(index) * 28)
                    }
                }
                .frame(width: 40 + 56, height: 48, alignment: .leading)
                VStack(alignment: .leading, spacing: 1) {
                    Text(photosSavedLabel)
                        .font(.system(size: 13, weight: .bold))
                        .foregroundStyle(Theme.Color.appText)
                    Text("Originals kept in your Vault")
                        .font(.system(size: 11.5, weight: .regular))
                        .foregroundStyle(Theme.Color.appTextSecondary)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                HStack(spacing: 3) {
                    Text("View")
                        .font(.system(size: 11.5, weight: .bold))
                    Icon(.chevronRight, size: 14, strokeWidth: 2.2, color: Theme.Color.categoryUnboxingDark)
                }
                .foregroundStyle(Theme.Color.categoryUnboxingDark)
            }
            .padding(Spacing.s3)
            .frame(maxWidth: .infinity)
            .background(Theme.Color.appSurface)
            .clipShape(RoundedRectangle(cornerRadius: Radii.xl, style: .continuous))
            .overlay(
                RoundedRectangle(cornerRadius: Radii.xl, style: .continuous)
                    .stroke(Theme.Color.appBorder, lineWidth: 1)
            )
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .accessibilityLabel("\(photosSavedLabel). Originals kept in your Vault. View photos.")
        .accessibilityIdentifier("unboxing_filedShots")
    }

    private var stackedThumb: some View {
        ZStack {
            Color(white: 0.11)
            CaptureStripeField()
        }
        .frame(width: 40, height: 48)
        .clipShape(RoundedRectangle(cornerRadius: Radii.sm, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: Radii.sm, style: .continuous)
                .stroke(Theme.Color.appSurface, lineWidth: 2)
        )
    }
}

// MARK: - ScanNextCard

@MainActor
struct ScanNextCard: View {
    let accent: Color
    let accentDark: Color
    let accentBg: Color
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: Spacing.s3) {
                RoundedRectangle(cornerRadius: Radii.md, style: .continuous)
                    .fill(accent)
                    .frame(width: 40, height: 40)
                    .overlay(Icon(.scanLine, size: 20, strokeWidth: 2.2, color: .white))
                    .shadow(color: accent.opacity(0.3), radius: 6, x: 0, y: 2)
                VStack(alignment: .leading, spacing: 1) {
                    Text("Scan the next item")
                        .font(.system(size: 14, weight: .bold))
                        .foregroundStyle(accentDark)
                    Text("Keep unboxing — capture flows back to here")
                        .font(.system(size: 12, weight: .regular))
                        .foregroundStyle(Theme.Color.appTextSecondary)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                Icon(.chevronRight, size: 18, strokeWidth: 2.2, color: accentDark)
            }
            .padding(Spacing.s4)
            .frame(maxWidth: .infinity)
            .background(accentBg)
            .clipShape(RoundedRectangle(cornerRadius: Radii.xl, style: .continuous))
            .overlay(
                RoundedRectangle(cornerRadius: Radii.xl, style: .continuous)
                    .stroke(accent.opacity(0.4), style: StrokeStyle(lineWidth: 1.5, dash: [5, 4]))
            )
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .accessibilityLabel("Scan the next item")
        .accessibilityIdentifier("unboxing_scanNext")
    }
}

/// Diagonal hairline stripes for the collapsed photo thumbnails — the
/// design's "never a hand-drawn object" placeholder fill. Mirrors the
/// `CameraScanner` stripe treatment so filed thumbs match the filmstrip.
private struct CaptureStripeField: View {
    var body: some View {
        GeometryReader { proxy in
            Path { path in
                let step: CGFloat = 7
                let total = proxy.size.width + proxy.size.height
                var x: CGFloat = -proxy.size.height
                while x < total {
                    path.move(to: CGPoint(x: x, y: 0))
                    path.addLine(to: CGPoint(x: x + proxy.size.height, y: proxy.size.height))
                    x += step
                }
            }
            .stroke(.white.opacity(0.07), lineWidth: 1)
        }
        .accessibilityHidden(true)
    }
}

File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/mailbox/unboxing/components/FiledSummary.kt
@file:Suppress("PackageNaming", "FunctionNaming", "MagicNumber", "LongMethod", "LongParameterList", "TooManyFunctions")

package app.pantopus.android.ui.screens.mailbox.unboxing.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.pantopus.android.ui.screens.mailbox.unboxing.UnboxingShot
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.PantopusIcon
import app.pantopus.android.ui.theme.PantopusIconImage
import app.pantopus.android.ui.theme.Radii
import app.pantopus.android.ui.theme.Spacing

/**
 * A17.14 filed-state chrome — the success [FiledBanner] ("Filed to Home ›
 * Warranties" + Undo) over the collapsed [FiledShots] photo summary ("4
 * photos saved · Originals kept in your Vault"). [ScanNextCard] is the
 * dashed launcher at the foot of the filed frame; the screen renders the AI
 * elf + locked `OcrFactsList` between them, matching the design order.
 * Mirrors iOS `FiledSummary`.
 */
@Composable
fun FiledSummary(
    filedTo: String,
    filedSubtitle: String,
    shots: List<UnboxingShot>,
    photosSavedLabel: String,
    onUndo: () -> Unit,
    onViewPhotos: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(modifier = modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(Spacing.s3)) {
        FiledBanner(filedTo = filedTo, filedSubtitle = filedSubtitle, onUndo = onUndo)
        FiledShots(shots = shots, photosSavedLabel = photosSavedLabel, onViewPhotos = onViewPhotos)
    }
}

@Composable
private fun FiledBanner(
    filedTo: String,
    filedSubtitle: String,
    onUndo: () -> Unit,
) {
    Row(
        modifier =
            Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.xl))
                .background(PantopusColors.successBg)
                .border(1.dp, PantopusColors.successLight, RoundedCornerShape(Radii.xl))
                .padding(horizontal = Spacing.s3, vertical = Spacing.s3)
                .testTag("unboxing_filedBanner"),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(Spacing.s3),
    ) {
        Box(
            modifier = Modifier.size(38.dp).clip(RoundedCornerShape(Radii.md)).background(PantopusColors.success),
            contentAlignment = Alignment.Center,
        ) {
            PantopusIconImage(
                icon = PantopusIcon.Check,
                contentDescription = null,
                size = 20.dp,
                strokeWidth = 2.4f,
                tint = PantopusColors.appTextInverse,
            )
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = "Filed to $filedTo",
                fontSize = 13.5.sp,
                fontWeight = FontWeight.Bold,
                color = PantopusColors.success,
            )
            Text(text = filedSubtitle, fontSize = 11.5.sp, color = PantopusColors.success)
        }
        Text(
            text = "Undo",
            fontSize = 10.sp,
            fontWeight = FontWeight.Bold,
            color = PantopusColors.success,
            modifier =
                Modifier
                    .clip(RoundedCornerShape(Radii.pill))
                    .background(PantopusColors.appSurface)
                    .border(1.dp, PantopusColors.successLight, RoundedCornerShape(Radii.pill))
                    .clickable(onClick = onUndo)
                    .padding(horizontal = Spacing.s2, vertical = 3.dp)
                    .testTag("unboxing_undo"),
        )
    }
}

@Composable
private fun FiledShots(
    shots: List<UnboxingShot>,
    photosSavedLabel: String,
    onViewPhotos: () -> Unit,
) {
    Row(
        modifier =
            Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.xl))
                .background(PantopusColors.appSurface)
                .border(1.dp, PantopusColors.appBorder, RoundedCornerShape(Radii.xl))
                .clickable(onClick = onViewPhotos)
                .padding(Spacing.s3)
                .semantics { contentDescription = "$photosSavedLabel. Originals kept in your Vault. View photos." }
                .testTag("unboxing_filedShots"),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(Spacing.s3),
    ) {
        Box(modifier = Modifier.width(96.dp).height(48.dp)) {
            shots.take(3).forEachIndexed { index, _ ->
                StackedThumb(modifier = Modifier.offset(x = (index * 28).dp))
            }
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(text = photosSavedLabel, fontSize = 13.sp, fontWeight = FontWeight.Bold, color = PantopusColors.appText)
            Text(
                text = "Originals kept in your Vault",
                fontSize = 11.5.sp,
                color = PantopusColors.appTextSecondary,
            )
        }
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(3.dp)) {
            Text(
                text = "View",
                fontSize = 11.5.sp,
                fontWeight = FontWeight.Bold,
                color = PantopusColors.categoryUnboxingDark,
            )
            PantopusIconImage(
                icon = PantopusIcon.ChevronRight,
                contentDescription = null,
                size = 14.dp,
                strokeWidth = 2.2f,
                tint = PantopusColors.categoryUnboxingDark,
            )
        }
    }
}

@Composable
private fun StackedThumb(modifier: Modifier = Modifier) {
    Box(
        modifier =
            modifier
                .size(width = 40.dp, height = 48.dp)
                .clip(RoundedCornerShape(Radii.sm))
                .background(Color(red = 0.11f, green = 0.11f, blue = 0.11f))
                .border(2.dp, PantopusColors.appSurface, RoundedCornerShape(Radii.sm)),
    ) {
        StripeField(modifier = Modifier.fillMaxSize())
    }
}

/**
 * A17.14 `ScanNext` launcher — the dashed teal card at the foot of the
 * filed frame. The Android border is solid (matching the `CameraScanner`
 * "Add" tile precedent); iOS dashes its stroke.
 */
@Composable
fun ScanNextCard(
    accent: Color,
    accentDark: Color,
    accentBg: Color,
    onTap: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier =
            modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.xl))
                .background(accentBg)
                .border(1.5.dp, accent.copy(alpha = 0.4f), RoundedCornerShape(Radii.xl))
                .clickable(onClick = onTap)
                .padding(Spacing.s4)
                .semantics { contentDescription = "Scan the next item" }
                .testTag("unboxing_scanNext"),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(Spacing.s3),
    ) {
        Box(
            modifier = Modifier.size(40.dp).clip(RoundedCornerShape(Radii.md)).background(accent),
            contentAlignment = Alignment.Center,
        ) {
            PantopusIconImage(
                icon = PantopusIcon.ScanLine,
                contentDescription = null,
                size = 20.dp,
                strokeWidth = 2.2f,
                tint = PantopusColors.appTextInverse,
            )
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(text = "Scan the next item", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = accentDark)
            Text(
                text = "Keep unboxing — capture flows back to here",
                fontSize = 12.sp,
                color = PantopusColors.appTextSecondary,
            )
        }
        PantopusIconImage(
            icon = PantopusIcon.ChevronRight,
            contentDescription = null,
            size = 18.dp,
            strokeWidth = 2.2f,
            tint = accentDark,
        )
    }
}

/**
 * Diagonal hairline stripes for the collapsed photo thumbnails — the
 * design's "never a hand-drawn object" placeholder fill, mirroring the
 * `CameraScanner` stripe treatment.
 */
@Composable
private fun StripeField(modifier: Modifier = Modifier) {
    Canvas(modifier = modifier) {
        val step = 7.dp.toPx()
        val color = Color.White.copy(alpha = 0.07f)
        var x = -size.height
        while (x < size.width + size.height) {
            drawLine(color, Offset(x, 0f), Offset(x + size.height, size.height), strokeWidth = 1f)
            x += step
        }
    }
}
